lib/repl.js

```diff
@@ -498,7 +506,8 @@ function REPLServer(prompt,
 
       // If error was SyntaxError and not JSON.parse error
       if (e) {
-        if (e instanceof Recoverable && !self.lineParser.shouldFail) {
+        if (e instanceof Recoverable && !self.lineParser.shouldFail &&
+            !sawCtrlD) {
```

Comment:
@thefourtheye `sawCtrlD` is set to `true` to indicate that current executed command is sourced from `editor` mode